workspaces/liferay-ticket-workspace/client-extensions/liferay-ticket-custom-element/src/components/TicketPreview.tsx

```diff
@@ -0,0 +1,184 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayButton from '@clayui/button';
+import ClayIcon from '@clayui/icon';
+import ClayLoadingIndicator from '@clayui/loading-indicator';
+import ClayPanel from '@clayui/panel';
+import {useDraggable} from '@dnd-kit/core';
+import {CSS} from '@dnd-kit/utilities';
+import {useContext, useMemo, useState} from 'react';
+import {QueryClient, useMutation} from 'react-query';
+
+import {TicketsAppContext} from '../context';
+import {Liferay} from '../services/liferay';
+import {assignTicketToMe} from '../services/tickets';
+import {Ticket} from '../types';
+
+const TicketPreview: React.FC<{ticket: Ticket}> = ({ticket}) => {
+	const [isLoading, setIsLoading] = useState(false);
+	const [isPanelExpanded, setIsPanelExpanded] = useState(false);
+	const queryClient: QueryClient = useContext(TicketsAppContext).queryClient;
+	const spritemap: string = useContext(TicketsAppContext).spriteMap;
+	const assigneeName = useMemo(
+		() => `${ticket.assignee?.givenName} ${ticket.assignee?.familyName}`,
+		[ticket.assignee]
+	);
+
+	const {
+		attributes,
+		isDragging,
+		listeners,
+		setNodeRef,
+		transform,
+	} = useDraggable({data: ticket, id: ticket.id + '_draggable'});
+
+	const style: React.CSSProperties = {
+		position: 'relative',
+		transform: CSS.Translate.toString(transform),
+		zIndex: isDragging ? 150 : 1,
+	};
+
+	const assignToMe = (ticket: Ticket) =>
+		new Promise((resolve, reject) => {
+			assignTicketToMe(ticket)
+				.then((result) => {
+					if (result.status === 200 || result.status === 204) {
+						return resolve(null);
+					}
+					else {
+						return reject();
+					}
+				})
+				.catch((error) => {
+					throw new Error(error);
```

Comment:
This doesn't seem like it would add an value... this is just catching an error and then throwing the same error, right?